src/traversals/randomwalks.jl

```diff
@@ -118,3 +118,64 @@ function self_avoiding_walk(g::AG, s::Integer, niter::Integer; seed::Int=-1) whe
     end
     return visited[1:(i - 1)]
 end
+
+"""
+    loop_erased_randomwalk(g, s, niter, distmx=weights(g); f=Set(), seed=-1, 
+                           rng=GLOBAL_RNG)
+
+Perform a [loop-erased random walk](https://en.wikipedia.org/wiki/Loop-erased_random_walk)
+on graph `g` starting at vertex `s` and continuing until one of the following
+conditions are met: (i) `niter` steps are performed, (ii) the path has no more 
+places to go, (iii) or the walk reaches an element in f.
+
+If f is specified and the final element of the walk is not in f, the function 
+will throw an error.
+
+Return a vector of vertices visited in order.
+"""
+function loop_erased_randomwalk(
+    g::AG, s::Integer, 
+    niter::Integer=max(100, nv(g)^2);
+    distmx::AbstractMatrix{T}=weights(g),
+    f::Union{Set{V},Vector{V}}=Set{Integer}(), 
+    seed::Int=-1,
+    rng::AbstractRNG=GLOBAL_RNG
+)::Vector{Int} where {T <: Real, U, AG <: AbstractGraph{U}, V <: Integer}
```

Comment:
If I remember `::SomeType` after a function does actually force `convert` to be used on the result of that function.